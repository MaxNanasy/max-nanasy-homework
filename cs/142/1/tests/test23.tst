const TEST = 12;void proc1() {	var x : int;	void proc11() {		var y : int;		type t = newType;		int proc111() {			return y;		}		int proc112() {			return ::proc111() + y;		}		y = ::proc111() + ::proc112();	}	void proc12() {		int proc121() {			return TEST;		}	}}main() {	::proc1();}