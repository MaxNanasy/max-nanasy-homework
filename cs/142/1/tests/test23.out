 CONST  ident  =  number  ;  VOID  ident  (  )  {  VAR  ident  :  INT  ;  VOID  ident  (  )  {  VAR  ident  :  INT  ;  TYPE  ident  =  ident  ;  INT  ident  (  )  {  RETURN  ident  ;  }  INT  ident  (  )  {  RETURN  :  :  ident  (  )  +  ident  ;  }  ident  =  :  :  ident  (  )  +  :  :  ident  (  )  ;  }  VOID  ident  (  )  {  INT  ident  (  )  {  RETURN  ident  ;  }  }  }  MAIN  (  )  {  :  :  ident  (  )  ;  }  <EOT> 